### PR TITLE
Add: a very basic CLI frontend for BaNaNaS API

### PR DESCRIPTION
**Pull request by TrueBrain**
It only allows you to list your own packages, and upload a new one.
And even that in a basic form. No dependencies, no conditions.

### CODE REVIEW

File: ape/commands/upload.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    status, data = await session.post(f"/new-package/{upload_token}/publish", json={})
+    if status == 400:
+        show_validation_errors(data)
+        raise Exit
+    if status != 201:
+        log.error(f"Server returned invalid status code {status}: {data}")
+        raise Exit
+
+    log.info("Uploaded successfull")
```
Comment (LordAro) [inline ape/commands/upload.py:85]:
```suggestion
    log.info("Uploaded successfully")
```

File: ape/session.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        response = await self.session.put(f"{self.api_url}{url}", json=json, headers=self._headers)
+        return await self._read_response(response)
+
+    def tus_upload(self, upload_token, fullpath, filename):
+        tus = TusClient(f"{self.tus_url}/new-package/tus/")
+        tus.set_headers({"Upload-Token": upload_token})
+        tus.set_headers({"Authorization": self._headers["Authorization"]})
+
+        try:
+            uploader = tus.uploader(fullpath, chunk_size=5 * 1024 * 1024, metadata={"filename": filename})
```
Comment (LordAro) [inline ape/session.py:64]: i'd personally prefer `chunk_size=(5 * 1024 * 1024)`, but if black is removing them, no issue...

File: README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+[![GitHub Tag](https://img.shields.io/github/v/tag/OpenTTD/bananas-frontend-cli?include_prereleases&label=stable)](https://github.com/OpenTTD/bananas-frontend-cli/releases)
+[![GitHub commits since latest release](https://img.shields.io/github/commits-since/OpenTTD/bananas-frontend-cli/latest/master)](https://github.com/OpenTTD/bananas-frontend-cli/commits/master)
+
+[![GitHub Workflow Status (Testing)](https://img.shields.io/github/workflow/status/OpenTTD/bananas-frontend-cli/Testing/master?label=master)](https://github.com/OpenTTD/bananas-frontend-cli/actions?query=workflow%3ATesting)
+
+This is a CLI frontend for the OpenTTD's content service, called BaNaNaS.
+
+## Development
+
+This API is written in Python 3.8 with aiohttp, and makes strong use of asyncio.
```
Comment (frosch123) [inline README.md:13]: Is this an API?

File: bananas_cli/authentication.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                    "client_id": "ape",
+                    "redirect_uri": "http://localhost:3977/",
+                    "code_verifier": Authenticate.code_verifier,
+                    "code": code,
+                    "grant_type": "authorization_code",
+                },
+            )
+            if status != 200:
+                print(data)
+                log.error(f"Server returned invalid status code {status}. Authentication failed.")
```
Comment (frosch123) [inline bananas_cli/authentication.py:47]: When does stuff go to print() and when to log?

File: README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+```bash
+python3 -m venv .env
+.env/bin/pip install -r requirements.txt
+```
+
+Next, check out the help.
+
+```bash
+.env/bin/python -m bananas_cli --help
+```
```
Comment (frosch123) [inline README.md:28]: Is click supposed to show description of the command args somewhere?
It does not for me.

Comment (frosch123) [reply README.md:28]: Ah, it does with "command -h". But for some reason it wants authentication / server connection just to print the help message.

**Comment by frosch123** [comment]
How much work do you want to put into parameter validation?

Most user errors result in a python backtrace, which scares people.
a) Invalid api-rul
b) Upload zero files.

**Comment by TrueBrain** [comment]
I think that is a very good point that should be addressed .. over time :D It is also missing a lot of functionality ;)

**Pull request merged**